Tools/build/generate_sbom.py

```diff
@@ -106,13 +108,106 @@ def filter_gitignored_paths(paths: list[str]) -> list[str]:
     return sorted([line.split()[-1] for line in git_check_ignore_lines if line.startswith("::")])
 
 
+def discover_pip_sbom_package(sbom_data: dict[str, typing.Any]) -> None:
+    """pip is a part of a packaging ecosystem (Python, surprise!) so it's actually
+    automatable to discover the metadata we need like the version and checksums
+    so let's do that on behalf of our friends at the PyPA.
+    """
+    global PACKAGE_TO_FILES, CPYTHON_ROOT_DIR
```

Comment:
We're not modifying `CPYTHON_ROOT_DIR`, does it need to be global?

```suggestion
    global PACKAGE_TO_FILES
```